File: MATLAB/push.m
function [Flows,excesses] = push( Capacities, Flows, excesses, u, v )
%push - operacja push - ustawia maksymalną wartość przedprzepływu
%na krawędzi wychodzącej z wierzchołka u i wchodzącej do wierzchołka v
send = min(excesses(u), Capacities(u,v) - Flows(u,v));
Flows(u,v) = Flows(u,v) + send;
Flows(v,u) = Flows(v,u) - send;
excesses(u) = excesses(u) - send;
excesses(v) = excesses(v) + send;
end

File: C++/push.cpp
#define MIN(X,Y) ((X) < (Y) ? (X) : (Y))

//operacja push - ustawia maksymalną wartość przedprzepływu
//na krawędzi wychodzącej z wierzchołka u i wchodzącej do wierzchołka v
void push(const int * const * C, int ** F, int *excess, int u, int v) {
  int send = MIN(excess[u], C[u][v] - F[u][v]);
  F[u][v] += send;
  F[v][u] -= send;
  excess[u] -= send;
  excess[v] += send;
}




